webview/src/setup/components/ProjectUninitialized.tsx

```diff
@@ -1,28 +1,83 @@
-import { VSCodeButton } from '@vscode/webview-ui-toolkit/react'
 import React from 'react'
 import { EmptyState } from '../../shared/components/emptyState/EmptyState'
+import { Button } from '../../shared/components/button/Button'
 
-interface ProjectUninitializedProps {
-  initializeProject: () => void
+const Header: React.FC = () => <h1>DVC is not initialized</h1>
+
+interface GitUninitializedProps {
+  canGitInitialize: boolean | undefined
+  initializeGit: () => void
 }
 
-export const ProjectUninitialized: React.FC<ProjectUninitializedProps> = ({
-  initializeProject
+const GitIsPrerequisite: React.FC = () => (
+  <p>A Git repository is a prerequisite of project initialization</p>
+)
+
+const GitUninitialized: React.FC<GitUninitializedProps> = ({
+  canGitInitialize,
+  initializeGit
 }) => {
-  return (
-    <EmptyState>
-      <div>
-        <h1>DVC is not initialized</h1>
+  if (!canGitInitialize) {
+    return (
+      <EmptyState>
+        <Header />
+        <GitIsPrerequisite />
         <p>
-          The current workspace does not contain a DVC project. You can
-          initialize a project which will enable features powered by DVC. To
-          learn more about how to use DVC please read{' '}
-          <a href="https://dvc.org/doc">our docs</a>
+          The extension is unable to initialize a Git repository in this
+          workspace
         </p>
-        <VSCodeButton onClick={initializeProject}>
-          Initialize Project
-        </VSCodeButton>
-      </div>
+        <p>
+          Please open a different folder which contains no Git repositories or a
+          single existing Git repository at the root
+        </p>
+      </EmptyState>
+    )
+  }
+
+  return (
+    <EmptyState>
+      <Header />
+      <GitIsPrerequisite />
+      <Button onClick={initializeGit} text="Initialize Git" />
     </EmptyState>
   )
 }
+
+const DvcUninitialized: React.FC<{ initializeDvc: () => void }> = ({
+  initializeDvc
+}) => (
+  <EmptyState>
+    <Header />
+    <p>
+      The current workspace does not contain a DVC project. You can initialize a
+      project which will enable features powered by DVC. To learn more about how
+      to use DVC please read <a href="https://dvc.org/doc">our docs</a>
+    </p>
+    <Button onClick={initializeDvc} text="Initialize Project"></Button>
+  </EmptyState>
+)
+
+export interface ProjectUninitializedProps {
+  canGitInitialize: boolean | undefined
+  initializeDvc: () => void
+  initializeGit: () => void
+  needsGitInitialized: boolean | undefined
+}
+
+export const ProjectUninitialized: React.FC<ProjectUninitializedProps> = ({
+  initializeDvc,
+  needsGitInitialized,
+  canGitInitialize,
+  initializeGit
+}) => {
+  if (needsGitInitialized) {
+    return (
+      <GitUninitialized
```

Comment:
Is there a way to detect if a git repo has been initialized via the CLI so the webview updates automatically: 

https://user-images.githubusercontent.com/43496356/209836990-c35cdb32-f818-4573-a4a9-86eaf09fd115.mov

Reply:
Yes, we need another watcher. I'll put it at the end of the chain as a follow-up.